extensions/wikia/Chat2/Chat.class.php

```diff
@@ -72,435 +82,207 @@ public static function echoCookies() {
 	 * messages (this is used primarily when the user is already banned from the wiki.
 	 * in that case, there is an error, but if the user is present they should be kicked).
 	 *
-	 * @param string $userNameToKickBan
-	 * @param User $kickingUser
+	 * @param string $subjectUserName
+	 * @param User $adminUser
 	 * @param int $time
 	 * @param string $reason
 	 *
 	 * @return bool|string Returns true on success, returns an error message as a string on failure.
 	 */
-	public static function banUser( $userNameToKickBan, $kickingUser, $time, $reason ) {
-		ChatHelper::info( __METHOD__ . ': Method called', [
-			'userNameToKickBan' => $userNameToKickBan,
-			'kickingUser' => $kickingUser,
+	public static function banUser( $subjectUserName, User $adminUser, $time, $reason ) {
```

Comment:
this function has around 60 lines, what about dividing it into separate functions and unit test them? For example:
- condition from line 103 to function `hasBanPermissions`
- action on add (lines 118-126)
- action on remove (unban) (l. 128-129) and so on